Gems/PhysX/Code/Source/BallJointComponent.cpp

```diff
@@ -46,11 +46,20 @@ namespace PhysX
 
         JointComponent::LeadFollowerInfo leadFollowerInfo;
         ObtainLeadFollowerInfo(leadFollowerInfo);
-        if (!leadFollowerInfo.m_followerActor)
+        if (leadFollowerInfo.m_followerActor == nullptr ||
+            leadFollowerInfo.m_followerBody == nullptr)
         {
             return;
         }
 
+        // if there is no lead body, this will be a constraint of the followers global position, so use invalid body handle.
+        AzPhysics::SimulatedBodyHandle parentHandle = AzPhysics::InvalidSimulatedBodyHandle;
+        if (leadFollowerInfo.m_leadBody != nullptr)
+        {
+            parentHandle = leadFollowerInfo.m_leadBody->m_bodyHandle;
+        }
+        AZ_Warning("PhysX", parentHandle == AzPhysics::InvalidSimulatedBodyHandle, "Entity [%s] Ball Joint component missing lead entity.", GetEntity()->GetName().c_str());
+
```

Comment:
This chunk of code is repeated multiple times. Maybe it could be moved to a function.

Reply:
i'll look into this in another PR there is more room for de-duplication that could be done with the joints code.